IPIP/0000-gateway-tar-response-format.md

```diff
@@ -0,0 +1,63 @@
+# IPIP 0000: Gateway TAR Response Format
+
+- Start Date: (format: 2022-10-10)
+- Related Issues:
+  - [ipfs/specs/pull/288](https://github.com/ipfs/specs/pull/288)
+  - [ipfs/go-ipfs/pull/9029](https://github.com/ipfs/go-ipfs/pull/9029)
+  - [ipfs/go-ipfs/pull/9034](https://github.com/ipfs/go-ipfs/pull/9034)
+
+## Summary
+
+Add TAR as a response format for the HTTP Gateway.
+
+## Motivation
+
+Currently, the HTTP Gateway only allows the download of single files, or
+CAR archives. However, CAR files are sometimes not necessary and user may
+want to download entire directories. An example use case is for the IPFS
+Web UI, where users are able to download files or directories.
+
+## Detailed design
+
+The solution is to allow the Gateway to support producing TAR archives
+by requesting them using either the `Accept` HTTP header or the `format`
+URL query.
+
+## Test fixtures
+
+Existing `curl` and `tar` tools can be used by implementers for testing.
+
+Providing static test vectors has little value here, as different TAR libraries may produce
+different byte-to-byte files  due to unspecified ordering of files and directories inside.
+
+## Design rationale
+
+The current gateway already supports different response formats via the
+`Accept` HTTP header and the `format` URL query. This RFC proposes adding
+one more supported format to that list.
+
+### User benefit
+
+Users will be able to directly download UnixFs directories from the gateway. In the Web UI,
+for example, we will be able to create a direct link to download the file, instead of using the
+API to put the whole file in memory before downloading it, saving resources and avoiding bugs.
+
+CLI users will be able to download a directory with existing tools like `curl` and `tar`.
```

Comment:
Does this put more strain on the gateway? @lidel

Reply:
As long it is uncompressed TAR, it will be _less_ strain than the current state: 
- no dynamic HTML with dir-index-html (current behavior) needs to be generated
- fewer requests need to be processed as user downloads entire directory in one run, instead of fetching each file individually.